### PR TITLE
[1.1.x] Apply feedrate to nozzle movement for kinematic machines

### PR DESCRIPTION
**Pull request by tcm0116**
#8778 repackaged for 1.1.x.

### CODE REVIEW

File: Marlin/planner.cpp [hunk 1]
```diff
@@ -1008,17 +1008,17 @@ void Planner::_buffer_steps(const int32_t (&target)[XYZE], float fr_mm_s, const
...
+    delta_mm[C_AXIS] = dc * steps_to_mm[C_AXIS];
   #endif
   delta_mm[E_AXIS] = esteps_float * steps_to_mm[E_AXIS_N];
 
-  if (block->steps[X_AXIS] < MIN_STEPS_PER_SEGMENT && block->steps[Y_AXIS] < MIN_STEPS_PER_SEGMENT && block->steps[Z_AXIS] < MIN_STEPS_PER_SEGMENT) {
+  if (block->steps[A_AXIS] < MIN_STEPS_PER_SEGMENT && block->steps[B_AXIS] < MIN_STEPS_PER_SEGMENT && block->steps[C_AXIS] < MIN_STEPS_PER_SEGMENT) {
     block->millimeters = FABS(delta_mm[E_AXIS]);
   }
-  else {
+  else if (NEAR_ZERO(millimeters)) {
```
Comment (thinkyhead) [inline Marlin/planner.cpp]: I'm sure we can just use `millimeters == 0.0` here.

File: Marlin/language_es.h [hunk 2]
```diff
@@ -115,9 +112,6 @@
 #define MSG_A_RETRACT                       _UxGT("Acel. retrac.")
 #define MSG_A_TRAVEL                        _UxGT("Acel. Viaje")
 #define MSG_STEPS_PER_MM                    _UxGT("Pasos/mm")
-#define MSG_XSTEPS                          _UxGT("X pasos/mm")
-#define MSG_YSTEPS                          _UxGT("Y pasos/mm")
-#define MSG_ZSTEPS                          _UxGT("Z pasos/mm")
```
Comment (thinkyhead) [inline Marlin/language_es.h]: I bet we can figure out many of these translations.

File: Marlin/planner.h [hunk 3]
```diff
@@ -372,7 +372,7 @@ class Planner {
 
       FORCE_INLINE static void unskew(float &cx, float &cy, const float &cz) {
         if (WITHIN(cx, X_MIN_POS, X_MAX_POS) && WITHIN(cy, Y_MIN_POS, Y_MAX_POS)) {
-          const float sx = cx + cy * xy_skew_factor + cz * xz_skew_factor,
+          const float sx = cx + cy * xy_skew_factor + cz * (xz_skew_factor - (xy_skew_factor * yz_skew_factor)),
```
Comment (thinkyhead) [inline Marlin/planner.h]: This turned out not to be correct. Originally I applied the fix in 368c124de8ffbb234ff8bb6c45ce0e1ab07e91da but then the author informed me it was wrong, and I double-checked to see it was true. So then f38f2ace5a82cd612ff699a3cd1617fe2ad16a3d followed two days later.

Comment (thinkyhead) [reply Marlin/planner.h]: See https://github.com/MarlinFirmware/Marlin/pull/8623#discussion_r157253068

**Comment by thinkyhead** [comment]
Looks good. I've updated the translations, reversed the [incorrect `skew_factor` change](https://github.com/MarlinFirmware/Marlin/pull/8623#discussion_r157253068), and changed `NEAR_ZERO(millimeters)` to just `!millimeters` in `_buffer_steps`.

I'll be interested to see how this affects my SCARA kinematics, which has to convert the feedrate from mm/s into a new degrees/s value on every segment.

File: Marlin/planner.h (hunk 3 above)
Comment (tcm0116) [reply Marlin/planner.h]: I think this was just an error in the merge

**Pull request merged**